docs/Building-functions.md

```diff
@@ -0,0 +1,111 @@
+# Building app action functions
+
+To start creating a function to be used in an App Action, use the CLI to generate a boilerplate including the basic things you'll need:
+
+```
+$ fliplet create-function com.example.function.say-hello "Say hello"
+```
+
+The above code will create a new folder named "say-hello" including the skeleton of your function, including these files:
+
+```
+js/build.js
+widget.json
+```
+
+## Including external libraries
+
+Functions can define the list of [dependencies](/Dependencies-and-assets.html#dependencies-and-assets) to be used in the `widget.json` file under the `"build" > "dependencies"` key of the file.
+
+## The function definition file
+
+The `widget.json` file defines your component as well as the **dependencies** and **assets** it needs in order to run.
+
+```json
+{
+  "name": "Say hello",
+  "package": "com.example.function.say-hello",
+  "description": "This function will create alert box with Hello message",
+  "version": "1.0.0",
+  "icon": "img/icon.png",
+  "tags": ["type:function"],
+   "interface": {
+    "dependencies": [],
+    "assets": []
+  },
+  "build": {
+    "dependencies": [],
+    "assets": ["js/build.js"],
+    "appAssets": []
+  },
+  "settings": {}
+}
+```
+
+---
+
+## Anatomy of a function
+
+Functions are very simple to create. They are just a single Javascript file that exports a function:
+
+```js
+Fliplet.Functions.register('com.example.function.say-hello', function(settings, context) {
+  alert('Hello');
+});
+```
+
+The above function will be executed when the function is called from the app. The function will receive two arguments:
+
+- `settings`: an object containing the settings of the function, as defined in the app action.
+- `context`: an object containing the shared context of the app action being run. This for example may include a `context.event` with the name of the event that triggered the function.
+
+The function can return a value, which will be passed to the app action as the result of the function. If the function returns a promise, the app action will wait for the promise to be resolved before continuing.
+
+Furthermore, the `context` object can be extended with data to be passed to the next function in the chain. For example, the following function will pass the `name` property to the next function in the chain:
+
+```js
+Fliplet.Functions.register('com.example.function.say-hello', function(settings, context) {
```

Comment:
I think this example is not great, given we have already defined a "say-hello" function above. Shall we use an example for a send Email function or a client side function? Our functions are stored in that functions gitguh repository you made so maybe we can take one from there>

Reply:
@debankurghosh2061 This example is simple one which explains how we can add additional properties to pass to the next function